detect/master_changer_test.go

```diff
@@ -0,0 +1,133 @@
+package detect
+
+import (
+	"reflect"
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesosphere/mesos-dns/logging"
+)
+
+func TestMasters_UpdatedMasters(t *testing.T) {
+	// create a new master changer with an unknown leader and no masters
```

Comment:
Since we renamed the type to `Masters` to read as `detect.Masters`, would it make sense to rename the files accordingly? Also, this comment.